aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -86,12 +81,6 @@ Tensor& randperm_out_cuda(Tensor& result, int64_t n, c10::optional<Generator> ge
 
   result.resize_({n});
 
-  if (n < 30000) {  // For small inputs, we offload it to CPU instead.
-    auto result_cpu = at::empty({n}, result.options().device(kCPU));
-    randperm_out(result_cpu, n, generator);
-    return result.copy_(result_cpu);
-  }
```

Comment:
The performance loss seems to be significant. Do you have a more concrete example in which these lines hurt?

Reply:
MaskRCNN uses randperm, and these lines prevent CUDA graph application of MaskRCNN